sass-msg-formatter.js

```diff
@@ -0,0 +1,105 @@
+const chalk = require('chalk');
+
+// colored text constants
+const errorText = chalk.bold.red;
+const warningText = chalk.yellow;
+const urlText = chalk.underline.cyan;
+const titleText = chalk.bgYellow;
+const numberedText = chalk.dim;
+const ruleNameText = chalk.bgWhite.black;
+
+/*
+ * filterForErrors: filters out elements in the results array that don't have errors
+ */
+function filterForErrors(result) {
+  return result.errored;
+}
+
+/*
+ * generateErrorIcon: generates a custom error or warning text depending on the severity of the error
+ */
+function generateErrorIcon(severity) {
+  let errorIcon = '';
+  switch (severity) {
+    case 'error':
+      errorIcon = errorText('ERROR');
+      break;
+    case 'warning':
+      errorIcon = warningText('warning');
+      break;
+    default:
+      errorIcon += '';
+      break;
+  }
+  return errorIcon;
+}
+
+/*
+ * formatTabbing: controls how many tabs are in between the location and the rule depending on how long the location is
+ */
+function formatTabbing(error) {
+  if (`${error.line}:${error.column}`.length < 6) {
+    return '\t \t';
+  }
+  return '\t';
+}
+
+/*
+ * createCustomMessage: returns a custom message with a link to the carbon documentation depending on what the error is
+ */
+function createCustomMessage(text) {
```

Comment:
could you add test coverage for this? 
also could you add jsdocs for the param?